### PR TITLE
[type:feat] Modify shenyu sdk and add shenyu.sdk.enable

### PR DESCRIPTION
**Pull request by yunlongn**
add `shenyu.sdk.enable` for shenyu-sdk

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Review by yu199195 — APPROVED** [review]

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-instance/src/main/java/org/apache/shenyu/springboot/starter/instance/ShenyuInstanceConfiguration.java [hunk 1]
```diff
@@ -39,4 +41,19 @@ public class ShenyuInstanceConfiguration {
...
+
+    /**
+     * ShenYu Instance Register Repository.
+     * expose to shenYu-sdk.
+     *
+     * @param config the config
+     * @return ShenYu Instance Register Repository
+     */
+    @Bean
+    @ConditionalOnProperty(name = "shenyu.instance.clientEnable", havingValue = "true")
```
Comment (yu199195) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-instance/src/main/java/org/apache/shenyu/springboot/starter/instance/ShenyuInstanceConfiguration.java:53]: whats mean？ 

ShenyuInstanceRegisterRepository already init in InstanceRegisterListener

Comment (yunlongn) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-instance/src/main/java/org/apache/shenyu/springboot/starter/instance/ShenyuInstanceConfiguration.java:53]: InstanceRegisterListener will register the service to the registry, shenyu-sdk does not need this operation.

**Review by yu199195 — CHANGES_REQUESTED** [review]
review

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-instance/src/main/java/org/apache/shenyu/springboot/starter/instance/ShenyuInstanceConfiguration.java (hunk 1 above)
Comment (yu199195) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-instance/src/main/java/org/apache/shenyu/springboot/starter/instance/ShenyuInstanceConfiguration.java:53]: Yes， so this code maybe in spring-boot-starter-sdk

Comment (yunlongn) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-instance/src/main/java/org/apache/shenyu/springboot/starter/instance/ShenyuInstanceConfiguration.java]: I get it, I'll revise it

**Review by yu199195 — APPROVED** [review]

**Pull request merged**